### ISSUE TITLE
unhandled error seperately when connecting with timeout

### ISSUE THREAD

**Issue by realshaft** [tediousjs/tedious#659]
I do have an issue with understanding this behaviour. Even if I listen in 'connected' to the error and cancel the connection, after a while (a minute or so, independent of the server timeout of 15s) the error event gets also called.

Qs: 
- If I cancel the connection in 'connected', why is there still being a error sent?
- Where does the second timeout comes from? when can I get sure that all asnc operations are done in tedious?

```
var connection = new Connection(config);

// Attempt to connect and execute queries if connection goes through
connection.on('connect', function(err) {
      if (err) {
        context.app.error(`Connection to ${xxx} not successful: ${err}`); // <--- Is executed so the error is handled
        connection.cancel();
      } else {
        context.app.info('Connected to ' + context.app.get('xxx').server);
});

// But after another timeout(?) the error event has been dispatched as well
connection.on('error', err => {
      context.app.error(`Connection to ${context.app.get('xxx').server} not successful: ${err}`);
});
```

**Comment by v-suhame** [comment]
Can you share the error you are getting in `connect` and `error` callback? 

Connection is usually closed after  `error` event is emitted, and call to `cancel` is executed only if there is a `request` pending. So, calling `connection.cancel();` while checking for error in `connect` event would return false.

**Comment by realshaft** [comment]
With the above code I do get this when the database is not reachable

in connection.on('connect'):
```
2017-12-13T13:35:20.753Z - error: Connection to 10.110.0.2 not successful: ConnectionError: Failed to connect to 10.110.0.2:11433 in 15000ms
2017-12-13T13:35:20.754Z - info: Connection to 10.110.0.2 closed
```
And as you can see after more than a minute I do get in connection.on('error') event as well:
```
2017-12-13T13:36:23.710Z - error: Connection to 10.110.0.2 not successful: ConnectionError: Connection lost - connect ETIMEDOUT 10.110.0.2:11433
```

**Comment by arthurschreiber** [comment]
This was fixed via #682. 👍

**Comment by kuseni** [comment]
I think this issue appears again in 2.6.4.

Code:
```javascript
log('started');

var conn = new Connection({
	server:   '8.8.8.8',
	userName: 'test',
	password: 'test'});

conn.on('connect', (err) => {
	log(`connect(): err="${err}"`);
	conn.close();
});

conn.on('end',    ()   => { log(`end()`); });
conn.on('error', (err) => { log(`error(): err="${err}"`); });
conn.on('debug', (err) => { log(`debug(): ${err}`); });
```
Log:
```
18:07:49.500: started
18:08:04.520: debug(): Failed to connect to 8.8.8.8:1433 in 15000ms
18:08:04.525: connect(): err="ConnectionError: Failed to connect to 8.8.8.8:1433 in 15000ms"
18:08:04.584: debug(): State change: Connecting -> Final
18:08:04.589: end()
18:08:10.529: debug(): Connection lost - connect ETIMEDOUT 8.8.8.8:1433
18:08:10.537: error(): err="ConnectionError: Connection lost - connect ETIMEDOUT 8.8.8.8:1433"
```
